### PR TITLE
Write tests for ticket queries

### PR DESCRIPTION
**Pull request by manzoorwanijk**
Completes #2100 

## Checklist

* [ ] I have read the documentation relating to systems affected by this pull request, see https://github.com/eventespresso/event-espresso-core/tree/master/docs
* [ ] User input is adequately validated and sanitized
* [ ] all publicly displayed strings are internationalized (usually using `esc_html__()`, see https://codex.wordpress.org/I18n_for_WordPress_Developers)
* [ ] My code is tested.
* [ ] My code follows the Event Espresso code style.
* [ ] My code has proper inline documentation.
* [ ] My code accounts for when the site is in Maintenance Mode (MM2 especially disallows usage of models)

### CODE REVIEW

File: assets/prototype/domain/eventEditor/data/queries/datetimes/test/useDatetimeIds.test.ts [hunk 1]
```diff
@@ -28,25 +24,10 @@ describe('useDatetimeIds()', () => {
...
-						espressoDatetimes: edge,
-					},
-				};
-				// write the test data to cache
-				client.writeQuery(writeQueryOptions);
-				// make sure the status flags is set for useDatetimes to work
-				useEffect(() => {
-					setIsLoaded(TypeName.datetimes, true);
-				}, []);
+				useInitDatetimeTestCache();
```
Comment (nicolad) [inline assets/prototype/domain/eventEditor/data/queries/datetimes/test/useDatetimeIds.test.ts]: Noice :+1:

File: assets/prototype/domain/eventEditor/data/queries/datetimes/test/useInitDatetimeTestCache.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { useApolloClient } from '@apollo/react-hooks';
+
+import useDatetimeQueryOptions from '../useDatetimeQueryOptions';
+import { ReadQueryOptions, WriteQueryOptions } from '../../../queries/types';
+import useInitDatetimeStatus from './useInitDatetimeStatus';
+import { edge } from './data';
+
+const useInitDatetimeTestCache = (espressoDatetimes = edge): void => {
```
Comment (nicolad) [inline assets/prototype/domain/eventEditor/data/queries/datetimes/test/useInitDatetimeTestCache.ts]: Handy :1st_place_medal:

File: assets/prototype/domain/eventEditor/data/queries/datetimes/test/useInitDatetimeStatus.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { useStatus, TypeName } from '../../../../../../application/services/apollo/status';
+
+
+const useInitDatetimeStatus = (): void => {
+	const { setIsLoaded, isLoaded } = useStatus();
+	if (!isLoaded(TypeName.datetimes)) {
+		setIsLoaded(TypeName.datetimes, true);
+	}
+};
```
Comment (nicolad) [inline assets/prototype/domain/eventEditor/data/queries/datetimes/test/useInitDatetimeStatus.ts]: It's good to have this here, not leaking this logic anywhere else.

Comment (manzoorwanijk) [reply assets/prototype/domain/eventEditor/data/queries/datetimes/test/useInitDatetimeStatus.ts]: Yes

**Pull request merged**